### DISCUSSION TITLE
Tailwind + Angular: @apply don't work

### DISCUSSION THREAD

**Discussion by GuidoABeltrame** [tailwindlabs/tailwindcss#4042] [Help]
Hello,
I am setting tailwind in Angular. Everything works perfect but when using @apply in some scss file it does not recognize it.
I give you my configuration of the project.

I would be very grateful if someone could help me with this.

Greetings!

My package.json

{
  "name": "oneway-frontend",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve & ngtw watch",
    "build": "ngtw build && ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "prestart": "ngtw build"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "~11.2.9",
    "@angular/common": "~11.2.9",
    "@angular/compiler": "~11.2.9",
    "@angular/core": "~11.2.9",
    "@angular/forms": "~11.2.9",
    "@angular/platform-browser": "~11.2.9",
    "@angular/platform-browser-dynamic": "~11.2.9",
    "@angular/router": "~11.2.9",
    "@fortawesome/angular-fontawesome": "^0.8.2",
    "@fortawesome/fontawesome-svg-core": "^1.2.34",
    "@fortawesome/free-brands-svg-icons": "^5.15.2",
    "@fortawesome/free-regular-svg-icons": "^5.15.2",
    "@fortawesome/free-solid-svg-icons": "^5.15.2",
    "@fullhuman/postcss-purgecss": "^4.0.3",
    "@ngxs/store": "^3.6.2",
    "@tailwindcss/forms": "^0.3.2",
    "@tailwindcss/typography": "^0.4.0",
    "font-awesome": "^4.7.0",
    "ng-tailwindcss": "^2.3.3",
    "ngx-toastr": "^13.2.1",
    "postcss-scss": "^3.0.5",
    "precss": "^4.0.0",
    "rxjs": "~6.6.0",
    "tslib": "^2.0.0",
    "zone.js": "~0.11.3"
  },
  "devDependencies": {
    "@angular-builders/custom-webpack": "^11.1.1",
    "@angular-devkit/build-angular": "~0.1102.8",
    "@angular/cli": "~11.2.8",
    "@angular/compiler-cli": "~11.2.9",
    "@ngxs/devtools-plugin": "^3.7.1",
    "@types/jasmine": "~3.6.0",
    "@types/node": "^12.11.1",
    "autoprefixer": "^10.2.5",
    "codelyzer": "^6.0.0",
    "jasmine-core": "~3.6.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~6.1.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.0.3",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "postcss": "^8.2.10",
    "postcss-import": "^14.0.1",
    "postcss-preset-env": "^6.7.0",
    "protractor": "~7.0.0",
    "tailwindcss": "^2.1.1",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~4.1.5"
  }
}

My angular.json styles:
            "styles": [
              "src/styles.scss",
              "node_modules/font-awesome/css/font-awesome.css",
              "node_modules/ngx-toastr/toastr.css"
            ],

My styles.scss
@import '../node_modules/tailwindcss/base';
@import '../node_modules/tailwindcss/components';
@import '../node_modules/tailwindcss/utilities';

My webpack.config:
const purgecss = require("@fullhuman/postcss-purgecss")({
  // Specify the paths to all of the template files in your project
  content: ["./src/**/*.html", "./src/**/*.component.ts"],
  // Include any special characters you're using in this regular expression
  defaultExtractor: (content) => content.match(/[\w-/:]+(?<!:)/g) || [],
});

module.exports = (config, options) => {
  console.log(`Using '${config.mode}' mode`);
  config.module.rules.push({
    test: /\.s[ac]ss$/i,
    use: [
      {
        loader: "postcss-loader",
        options: {
          postcssOptions: {
            plugins: [
              require("tailwindcss")("./tailwind.config.js"),
              require("autoprefixer"),
              ...(config.mode === "production" ? [purgecss] : []),
            ],
          }
        },
      },
    ],
  });
  return config;
};

My tailwind.config.js
module.exports = {
  prefix: '',
  purge: {
    enabled: process.env.NODE_ENV === 'production',
    content: [
      './src/**/*.{html,ts}',
    ]
  },
  darkMode: 'class', // or 'media' or 'class'
  theme: {
    extend: {
      colors: {
        primary: '#F2802E',
        secondary: '#3764FB',
        success: '#4A9937',
        warning: '#F5222D',
        info: '#108EE9',
      }
    }
  },
  variants: {
    extend: {},
  },
  plugins: [require('@tailwindcss/forms'), require('@tailwindcss/typography')],
};

**Comment by simonswiss** [comment]
Hey!

Usually, this happens when the CSS runs through the Tailwind plugin _before_ the `scss` is processed. Try and make sure the `scss` styles are compiled first, and then that output is going through PostCSS.

Not sure how exactly the pipeline works in Angular, but I am under the impression the issue is around the `angular.json => styles` array:

```js
// My angular.json styles:
"styles": [
  "src/styles.scss",
  "node_modules/font-awesome/css/font-awesome.css",
  "node_modules/ngx-toastr/toastr.css"
],
```

```css
/* My styles.scss */
@import '../node_modules/tailwindcss/base';
@import '../node_modules/tailwindcss/components';
@import '../node_modules/tailwindcss/utilities';
```

What happens if you write the `@apply` declarations inside `src/styles.scss` instead of a separate `.scss` file?
